tools/docker/wait-for-postgres.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+RETRIES=5
+THRESHOLD=2
```

Comment:
What does the threshold do?  From what I understand, its decremented when the postgres database is ready.  I am a little confused why we need to check this multiple times, but I am probably misinterpreting this.

Reply:
You're correct. We need to do this because sometimes when the postgres database is started again or initializing, it'll initialize (so it'll also accept connections) and then restart again. If you start running queries when it's first available, they might fail.

Reply:
Thanks for the explanation, maybe it would be good to add a note about this in the code?